src/meta/rule.schema.json

```diff
@@ -1,150 +1,464 @@
+  "definitions": {
+    "Then": {
+      "allOf": [
+        {
+          "properties": {
+            "field": {
+              "type": "string"
+            }
+          }
+        },
+
+        {
+          "if": {
+           "properties": {
+             "function": {
+               "const": "alphabetical"
+             }
+           }
+          },
+          "then": {
             "properties": {
-                "field": {
+              "functionOptions": {
+                "additionalProperties": false,
+                "properties": {
+                  "keyedBy": {
                     "type": "string"
+                  }
                 },
-                "function": {
-                    "$ref": "#/definitions/T"
+                "type": ["object", "null"]
+              }
+            }
+          }
+        },
+
+        {
+          "if": {
+            "properties": {
+              "function": {
+                "const": "casing"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "functionOptions": {
+                "additionalProperties": false,
+                "properties": {
+                  "type": {
+                    "type": "string",
+                    "enum": ["flat", "camel", "pascal", "kebab", "cobol", "snake", "macro"]
+                  },
+                  "disallowDigits": {
+                    "type": "boolean"
+                  },
+                  "separator": {
+                    "additionalProperties": false,
+                    "properties": {
+                      "char": {
+                        "type": "string",
+                        "maxLength": 1
+                      },
+                      "allowLeading": {
+                        "type": "boolean"
+                      }
+                    },
+                    "required": ["char"],
+                    "type": "object"
+                  }
                 },
-                "functionOptions": {
-                    "$ref": "#/definitions/O"
-                }
+                "required": ["type"],
+                "type": "object"
+              }
             },
-            "required": [
-                "function"
-            ],
-            "type": "object"
-        },
-        "O": {
-            "type": "object"
+            "required": ["functionOptions"]
+          }
         },
-        "T": {
-            "type": "string"
+
+        {
+          "if": {
+            "properties": {
+              "function": {
+                "const": "enumeration"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "functionOptions": {
+                "additionalProperties": false,
+                "properties": {
+                  "values": {
+                    "type": "array",
+                    "items": {
+                      "type": ["string", "number"]
```

Comment:
@P0lip Note: Might be worth extending the doco on the restricted types

Reply:
Which documentation you mean?
The one about built-in functions?

Reply:
@P0lip Yes. This was what I was thinking about